### PR TITLE
Fixes Reference Summed Potential

### PR DESCRIPTION
**Pull request by badisa**
* Only supports 2 potentials in reference summed potential right now

### CODE REVIEW

**Comment by mcwitt** [comment]
Good catch! The fix looks correct to me.

File: tests/test_potentials.py [hunk 1]
```diff
@@ -263,6 +263,23 @@ def test_summed_potential(harmonic_bond_test_system):
         GradientTest().compare_forces_gpu_vs_reference(coords, [params], box, potential, rtol, precision)
 
 
+@pytest.mark.parametrize("num_potentials", [1, 2, 5])
+def test_summed_potential_supports_arbitrary_number_of_potentials(num_potentials, harmonic_bond_test_system):
+    """Verifies that summed potential can handle various numbers of potentials. Previously bug
+    only allowed 1 or 2 potentials."""
```
Comment (mcwitt) [inline tests/test_potentials.py:269]: Just to verify: the bug only affected the reference summed potential, and had not been exposed up until now; correct?

Comment (badisa) [reply tests/test_potentials.py:269]: Correct

Comment (mcwitt) [reply tests/test_potentials.py:269]: nit: any reason not to merge this into the above `test_summed_potential`, strengthening it to test arbitrary numbers of potentials?

Comment (badisa) [reply tests/test_potentials.py:269]: Clarified in docstring https://github.com/proteneer/timemachine/pull/930/commits/fb60b4973149dbeb044a8a4d4772f89e79c4e85a

File: timemachine/potentials/summed.py [hunk 2]
```diff
@@ -37,5 +37,7 @@ def summed_potential(
     """
     assert len(U_fns) == len(shapes)
     sizes = np.prod(shapes, axis=1)
-    paramss = [ps.reshape(shape) for ps, shape in zip(np.split(params, sizes[:-1]), shapes)]
+    # np.split expects indices, must increment sizes to be indices
+    split_indices = np.cumsum(sizes)
+    paramss = [ps.reshape(shape) for ps, shape in zip(np.split(params, split_indices[:-1]), shapes)]
```
Comment (mcwitt) [inline timemachine/potentials/summed.py:42]: This looks correct to me.

File: tests/test_potentials.py (hunk 1 above)
Comment (badisa) [reply tests/test_potentials.py:269]: > nit: any reason not to merge this into the above `test_summed_potential`, strengthening it to test arbitrary numbers of potentials?

My reasoning was that the other test is testing the consistency of the test system, while this is testing that arbitrary numbers of potentials can be combined. Will merge them back together.

Comment (mcwitt) [reply tests/test_potentials.py:269]: > Will merge them back together.

I have a slight preference for this, but non-blocking if you have a reason to prefer them separate.

**Review by mcwitt — APPROVED** [review]

**Review by maxentile — APPROVED** [review]

**Pull request merged**